File: javascript/test/Behavioral/Command.js
(function() {
  define(['Behavioral/Command/DvdName', 'Behavioral/Command/DvdCommandNameStarsOn', 'Behavioral/Command/DvdCommandNameStarsOff'], function(DvdName, DvdCommandNameStarsOn, DvdCommandNameStarsOff) {
    'use strict';
    return describe('Test Behavioral Command', function() {
      var bobStarsOff, bobStarsOn, jayAndBob, spongeBob, spongeStarsOff, spongeStarsOn;
      jayAndBob = new DvdName('Jay and Silent Bob Strike Back');
      spongeBob = new DvdName('Sponge Bob Squarepants - Nautical Nonsense and Sponge Buddies');
      bobStarsOn = new DvdCommandNameStarsOn(jayAndBob);
      bobStarsOff = new DvdCommandNameStarsOff(jayAndBob);
      spongeStarsOn = new DvdCommandNameStarsOn(spongeBob);
      spongeStarsOff = new DvdCommandNameStarsOff(spongeBob);
      it('as first instantiated', function() {
        expect(jayAndBob.toString()).toEqual('DVD: Jay and Silent Bob Strike Back');
        expect(spongeBob.toString()).toEqual('DVD: Sponge Bob Squarepants - Nautical Nonsense and Sponge Buddies');
        return expect(JSON.stringify(spongeStarsOn)).toEqual('{"dvdName":{"titleName":"Sponge Bob Squarepants - Nautical Nonsense and Sponge Buddies"}}');
      });
      it('stars on', function() {
        bobStarsOn.execute();
        spongeStarsOn.execute();
        expect(jayAndBob.toString()).toEqual('DVD: Jay*and Silent Bob Strike Back');
        return expect(spongeBob.toString()).toEqual('DVD: Sponge*Bob Squarepants - Nautical Nonsense and Sponge Buddies');
      });
      return it('sponge stars off', function() {
        spongeStarsOff.execute();
        expect(jayAndBob.toString()).toEqual('DVD: Jay*and Silent Bob Strike Back');
        return expect(spongeBob.toString()).toEqual('DVD: Sponge Bob Squarepants - Nautical Nonsense and Sponge Buddies');
      });
    });
  });

}).call(this);

File: java/t/test/Behavioral/Command.java
//TestCommand.java - testing the Command


package test.Behavioral;


import static org.junit.Assert.assertEquals;

import org.junit.Test;
import org.junit.Ignore;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

import Behavioral.Command.*;


class Command {            

	@Test
   public void comtest() {
		
       DvdName jayAndBob = new DvdName("Jay and Silent Bob Strike Back");
       DvdName spongeBob = new DvdName("Sponge Bob Squarepants - Nautical Nonsense and Sponge Buddies");

       CommandAbstract bobStarsOn = new DvdCommandNameStarsOn(jayAndBob);
       CommandAbstract bobStarsOff = new DvdCommandNameStarsOff(jayAndBob);
       CommandAbstract spongeStarsOn = new DvdCommandNameStarsOn(spongeBob);
       CommandAbstract spongeStarsOff = new DvdCommandNameStarsOff(spongeBob);

       bobStarsOn.execute();
       spongeStarsOn.execute();

		org.junit.Assert.assertEquals( "stars on", 
			"DVD: Jay*and*Silent*Bob*Strike*Back",
			jayAndBob.toString()
		);
		org.junit.Assert.assertEquals( "stars on", 
			"DVD: Sponge*Bob*Squarepants*-*Nautical*Nonsense*and*Sponge*Buddies",
			spongeBob.toString()
		);

		spongeStarsOff.execute();
		org.junit.Assert.assertEquals( "stars off on just spongebob, still on jay and bob ",
			"DVD: Jay*and*Silent*Bob*Strike*Back",
			jayAndBob.toString()
		);
		org.junit.Assert.assertEquals( "stars off on just spongebob", 
			"DVD: Sponge Bob Squarepants - Nautical Nonsense and Sponge Buddies",
			spongeBob.toString()
		);

		
   }
}      
